rpcserver_test.go

```diff
@@ -165,3 +174,36 @@ func TestRpcServer(t *testing.T) {
 		currentTestNum++
 	}
 }
+
+func TestCertCreationWithHosts(t *testing.T) {
+	certfile, _ := ioutil.TempFile("", "certfile")
```

Comment:
I'd suggest checking errors using `t.Fatalf` to display them and fail the test if non nil.  Otherwise, it will be difficulty to figure out why the test is failing.

Same goes for the rest.